### PR TITLE
Adding GarNet examples + config-files and data directories

### PR DESCRIPTION
**Pull request by yiiyama**
~~To be merged after hls4ml [PR 213](https://github.com/hls-fpga-machine-learning/hls4ml/pull/213) is merged.~~
Although the examples would not convert / build without hls4ml [PR 213](https://github.com/hls-fpga-machine-learning/hls4ml/pull/213), let's merge this PR first because 213 uses one of the new examples for build tests.

### CODE REVIEW

**Comment by yiiyama** [comment]
Or perhaps right before it is merged, because I'm adding a test line that uses the 1layer model.

**Comment by Duchstf** [comment]
Hm ... this actually might complicate things a little bit. Because of a few things:

1. The current `example-models` submodule in `hls4ml` is tagged at [an older commit](https://github.com/hls-fpga-machine-learning/example-models/tree/5f7c636ff5ed90d4d37710dc46b9e5344c4c057d). So if you want to add the test you need to move the tag to the latest `example-models` version in master. 

2. However, it might mess up other tests, since there have been some minor changes in model file names here. But I'll look into this, I think it should be easy to fix. Do you know when the Garnet PR is planned to be merged?

**Comment by Duchstf** [comment]
Actually, regarding point #2, I don't think there would be any problem now that I looked into the test. So you just have to move the tag 😁. 

Also, maybe we can merge this PR first, and then I'll make some changes to the fetching examples scheme (should be quick) to your PR 213, then we can merge it to `hls4ml`. But it depends on your timeline, if you prefer to merge PR 213 before being able to fetch the garnet examples, then it's fine too. I might just make a separate PR after yours to support downloading them.

**Comment by yiiyama** [comment]
Thanks for the comments @Duchstf ! I think it's better to merge this PR first before 213 (despite my original comment in this PR - which I edited now).

File: config-files/garnet_3layer_config.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+KerasJson: garnet_3layer.json
+KerasH5:   garnet_3layer_weights.h5
```
Comment (Duchstf) [inline config-files/garnet_3layer_config.yml:2]: I think everything looks good now. I just have a question here. If you do end up using the config file in your test, wouldn't you use relative path here? Or is it not important?

Comment (Duchstf) [reply config-files/garnet_3layer_config.yml:2]: If you are fine leaving them like this I can merge this PR now.

Comment (yiiyama) [reply config-files/garnet_3layer_config.yml:2]: Yeah I think it's fine like this. These lines are sort of suggesting that the user should copy all relevant files into a work directory before running `hls4ml convert`.

**Pull request merged**